test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala

```diff
@@ -432,6 +433,22 @@ class OptimizerTest {
     assertEquals("Scala.js", "Scala.j" + 's')
   }
 
+  // Division by zero
+
+  @Test def divideByZero_Issue4604(): Unit = {
+    @noinline def divByZeroInExpressionPosition(): Int = {
+      0 / 0
+    }
+
+    @noinline def divByZeroInStatementPosition(): Unit = {
+      0 / 0
+      fail("should be unreachable")
+    }
```

Comment:
Also add a test case for Longs?

Reply:
Ah yes, of course. I should have thought about that. Done.